hudi-common/src/main/java/org/apache/hudi/common/table/timeline/versioning/v2/ActiveTimelineV2.java

```diff
@@ -526,12 +560,14 @@ protected <T> void transitionStateToComplete(boolean shouldLock, HoodieInstant f
           throw new HoodieIOException(
               "Could not rename " + fromInstantPath + " to " + toInstantPath);
         }
+        return instantWithCompletionTime;
       } else {
         // Ensures old state exists in timeline
         ValidationUtils.checkArgument(
             metaClient.getStorage().exists(getInstantFileNamePath(fromInstantFileName)),
             "File " + getInstantFileNamePath(fromInstantFileName) + " does not exist!");
-        createCompleteFileInMetaPath(shouldLock, toInstant, metadata);
+        String completionTime = HoodieInstantTimeGenerator.formatDateUTC(new Date(createCompleteFileInMetaPath(shouldLock, toInstant, metadata)));
```

Comment:
nts: is this compatible with other changes we are doing to smoothen time generation